pkg/collectors/infrastructure.go

```diff
@@ -0,0 +1,104 @@
+package collectors
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kube-state-metrics/pkg/metric"
+	"k8s.io/kube-state-metrics/pkg/version"
+
+	"github.com/golang/glog"
+
+	v1 "github.com/openshift/api/config/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+)
+
+var (
+	descConfigLabelsDefaultLabels = []string{"namespace", "config"}
```

Comment:
You should have a separate variable per type:
Infrastructure: type
Proxies: type
FeatureSets: name

Reply:
Actually you may not need a set of default labels for infrastructure, given the type and region are determined in the GenerateFunc